zml/hostbuffer.zig

```diff
@@ -251,6 +256,20 @@ pub const HostBuffer = struct {
         };
     }
 
+    pub fn awaitt(self: *HostBuffer) !*HostBuffer {
+        if (self._ready) {
+            return self;
+        }
+        if (self._event) |ev| {
```

Comment:
this logic would be simpler if we used the "make invalid state unrepresentable" strategy:

`ready: union { ready, ev: struct { *Event, *Api } };`

Reply:
👍 great suggestion!